util/postprocess-makedepend.pl

```diff
@@ -0,0 +1,160 @@
+#! /usr/bin/env perl
+# Copyright 2018 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the OpenSSL license (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+
+use strict;
+use warnings;
+
+use lib '.';
+use configdata;
+
+use File::Spec::Functions qw(canonpath rel2abs);
+
+my $abs_srcdir = rel2abs($config{sourcedir});
+my $abs_blddir = rel2abs($config{builddir});
+
+my $producer = shift @ARGV;
+die "Producer not given\n" unless $producer;
+
+my $procedure = {
+    'makedepend' =>
+        sub {
+            my $line = shift;
+
+            # makedepend, in its infinite wisdom, wants to have the object file
+            # in the same directory as the source file.  This doesn't work too
+            # well with out-of-source-tree builds, so we must resort to tricks
+            # to get things right.  The trick is to call makedepend with an
+            # extra suffix that contains the desired object file path, like
+            # this:
+            #
+            #   makedepend -f- -o"|dir/foo.o" -- $(CFLAGS) -- ../some/foo.c
+            #
+            # The result will look something like this:
+            #
+            #   ../somewhere/foo|dir/foo.o: deps...
+            #
+            # Which is easy to massage by removing everything up to the first |
+
+            # Remove everything up to the first |
+            $line =~ s/^.*\|//;
+            # Also, remove any dependency that starts with a /, because those
+            # are typically system headers
```

Comment:
But it doesn't list system headers... So in a sense it's a misleading comment... Maybe it would appropriate to add "in case they are added"...

Reply:
Err, say what? `makedepend` does list system headers too, see the example I gave above...

Reply:
Hmmmm, mine didn't... At least I didn't see it when commented, but then it was late... OK, disregard this for now.

Reply:
Did you happen to have added `-Y` when you tried?  I just discovered that flag, and sure enough, then there are no system headers...

Reply:
Just in case, the question that triggered me to look into it was "what if user uses absolute pathname to Configure". I mean I [for one] am accustomed to `mkdir foo; cd foo; ..\openssl\config; make`, but just might be just me, as I've met people who prefer absolute path names. Maybe it's appropriate to recognize that dependencies are not that meaningful to those who just build and use OpenSSL. They are meaningful only to active OpenSSL developers. And they probably manage, because they are more likely to use gcc. In other words one can as well say that it wouldn't be inappropriate to leave it as it is now.